modules/nosql/manager_redis.go

```diff
@@ -59,8 +60,84 @@ func (m *Manager) GetRedisClient(connection string) redis.UniversalClient {
 		name: []string{connection, uri.String()},
 	}
 
+	opts := getRedisOptions(uri)
+	tlsConfig := getRedisTLSOptions(uri)
+
+	clientName := uri.Query()["clientname"]
+
+	if clientName != nil && len(clientName) > 0 {
+		client.name = append(client.name, clientName[0])
+	}
+
+	switch uri.Scheme {
+	case "redis+sentinels":
+		fallthrough
+	case "rediss+sentinel":
+		opts.TLSConfig = tlsConfig
+		fallthrough
+	case "redis+sentinel":
+		if uri.Host != "" {
+			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
+		}
```

Comment:
Add a error log when Host is empty? 

And the code for below `"redis+cluster"` and `"redis"`  are really similar, maybe they can be extracted into a helper function

Reply:
Same as above -- I'm happy to do some more refactoring here, but this is all the original code, just pulled into a method for testing. Let me know what you'd like!